[BUG] Setup.sh not working on macos 26.4.1

I was trying to compile on macOS 26.4.1. It seems the removing of linking AGL framework is incompleted, which triggered errors. I feel like the linking to AGL lib is committed by Qt lib.

: && /usr/bin/c++ -O3 -DNDEBUG -arch arm64 -mmacosx-version-min=11.0 -dynamiclib -Wl,-headerpad_max_install_names -o
_deps/qgeoview-build/lib/libqgeoview.dylib -install_name @rpath/libqgeoview.dylib
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/qgeoview_autogen/mocs_compilation.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVUtils.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVGlobal.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVProjection.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVProjectionEPSG3857.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVCamera.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVMap.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVMapQGItem.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVMapQGView.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVMapRubberBand.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVItem.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVDrawItem.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayer.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerTiles.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerTilesOnline.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerGoogle.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerBing.cpp.o
 _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerOSM.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVLayerBDGEx.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVWidget.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVWidgetCompass.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVWidgetScale.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVWidgetZoom.cpp.o _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/QGVWidgetText.cpp.o
 _deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/Raster/QGVImage.cpp.o
_deps/qgeoview-build/lib/CMakeFiles/qgeoview.dir/src/Raster/QGVIcon.cpp.o -F/Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib
-Wl,-rpath,/Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib
/Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib/QtWidgets.framework/Versions/A/QtWidgets
/Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib/QtNetwork.framework/Versions/A/QtNetwork
/Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib/QtGui.framework/Versions/A/QtGui  -framework AGL  -framework AppKit  -framework OpenGL
-framework ImageIO  -framework Metal  /Users/gluo/FinceptTerminal/.qt/6.8.3/clang_64/lib/QtCore.framework/Versions/A/QtCore  -framework IOKit
-framework DiskArbitration  -framework UniformTypeIdentifiers && :
ld: framework 'AGL' not found
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[2/323] Linking CXX static library _deps/qxlsx-build/libQXlsxQt6.a
ninja: build stopped: subcommand failed.
  ERROR: Build failed. See error above.

--- a/fincept-qt/CMakeLists.txt
+++ b/fincept-qt/CMakeLists.txt
@@ -127,6 +127,27 @@ option(FINCEPT_BUILD_TESTS "Build the Fincept Terminal test suite" OFF)
 # installer. Developers may override for local experimentation with
 # -DFINCEPT_ALLOW_QT_DRIFT=ON (not for CI or release builds).
 set(FINCEPT_QT_VERSION 6.8.3)
+
+# Qt 6.8 FindWrapOpenGL.cmake hard-codes "-framework AGL" as a fallback when
+# find_library(AGL) fails. Apple removed AGL in macOS 14+ (Sonoma), causing
+# linker errors on modern SDKs. Pre-create WrapOpenGL::WrapOpenGL without AGL
+# so Qt's FindWrapOpenGL takes its early-return path (target already exists).
+if(APPLE AND NOT TARGET WrapOpenGL::WrapOpenGL)
+    find_package(OpenGL QUIET)
+    if(OpenGL_FOUND)
+        add_library(WrapOpenGL::WrapOpenGL INTERFACE IMPORTED)
+        get_target_property(_fincept_ogl_loc OpenGL::GL IMPORTED_LOCATION)
+        if(_fincept_ogl_loc AND NOT _fincept_ogl_loc MATCHES "/([^/]+)\\.framework$")
+            get_filename_component(_fincept_ogl_path "${_fincept_ogl_loc}" DIRECTORY)
+        endif()
+        if(NOT _fincept_ogl_path)
+            set(_fincept_ogl_path "-framework OpenGL")
+        endif()
+        target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE "${_fincept_ogl_path}")
+        set(WrapOpenGL_FOUND ON)
+    endif()
+endif()
+
 if(FINCEPT_ALLOW_QT_DRIFT)
     find_package(Qt6 REQUIRED COMPONENTS
         Widgets Charts PrintSupport Network Sql Concurrent Multimedia)

